packages/data/src/navigation/reducer.js

```diff
@@ -25,6 +26,12 @@ const reducer = (
 				menuItems: [ ...state.menuItems, ...menuItems ],
 			};
 			break;
+		case TYPES.SET_PERSISTED_QUERY:
```

Comment:
I think we can forego the `setPersistedQuery` entirely and use the events directly here instead of creating a setter.

```suggestion
		case TYPES.ON_HISTORY_CHANGE:
		case TYPES.ON_LOAD:
```

Reply:
Good call, done in d103563869918ec0613e2cd9f21dcf754e032b32